ArcGISRuntimeSDKQt_CppSamples/Layers/ListKmlContents/ListKmlContents.cpp

```diff
@@ -231,8 +197,16 @@ void ListKmlContents::processSelectedNode(const QString& nodeName)
         m_sceneView->setViewpoint(Viewpoint(nodeExtent));
       }
 
-      // show the children of the node
-      displayChildren(node);
+      // reset m_lastLevel before levelNodeNames() is called
+      m_lastLevel = false;
+      emit levelNodeNamesChanged();
+
+      // if displaying end-nodes, change m_currentNode to first end-node for correct behavior of back button
+      if (m_lastLevel)
+      {
+        m_currentNode = dynamic_cast<KmlContainer*>(m_currentNode)->childNodesListModel()->at(0);
```

Comment:
If we can guarantee this is a container, we can use `static_cast` which is faster at runtime. Otherwise we should check if the cast returns a `nullptr` or not!